modules/common/pod/pod.go

```diff
@@ -47,3 +47,21 @@ func GetPodListWithLabel(
 
 	return podList, nil
 }
+
+// GetHeadlessPodFQDN gets a list of pods matching the labels provided and returns a slice of pod FQDNs. This
+// is required in order to allow resolution of pods behind Headless services.
+func GetHeadlessPodFQDN(ctx context.Context, h *helper.Helper, serviceName string, namespace string, labelSelector map[string]string) ([]string, error) {
+	var podSvcNames []string
+	var podList *corev1.PodList
+
+	podList, err := GetPodListWithLabel(ctx, h, namespace, labelSelector)
+	if err != nil {
+		return []string{}, fmt.Errorf("Error getting list of pods: %v", err)
```

Comment:
please use %w to wrap the original error into the new error. https://go.dev/blog/go1.13-errors

Reply:
Done